librz/cons/cons.c

```diff
@@ -1023,6 +1025,18 @@ RZ_API void rz_cons_flush(void) {
 		__cons_write(I.context->buffer, I.context->buffer_len);
 	}
 
+	if (I.context->buffer && I.context->buffer_len) {
+		I.context->buffer[I.context->buffer_len] = 0;
+		const char *p = &I.context->buffer[I.context->buffer_len - 1];
+		for (; *p != '\n' && p > I.context->buffer; p--)
+			;
+		if (*p == '\n') {
+			RZ_FREE(I.context->last_nonl_line);
+			p++;
+		}
+		I.context->last_nonl_line = rz_str_append(I.context->last_nonl_line, p);
+	}
+
```

Comment:
Is this really necessary? Can't we force the output by adding a `rz_cons_printf("\n");` in the `aezsu` command? I think it is better than having rz_cons automagically perform this.

Reply:
> Is this really necessary?

Strictly speaking, for `aeszu` yes an extra newline is just an inaccuracy, but suppose a buggy command produces output that is not terminated with a newline. The current code will just make the final line disappear and you'd be lucky if you realize what was happening.

> Can't we force the output by adding a `rz_cons_printf("\n");` in the `aezsu` command?

Yes but if the emulated code does actually print a newline then there will be a double newline. The only way `aezsu` can determine whether a newline is needed or not is to check what the final character of the output is, and for that you need code in rz_cons that is probably similar to the above.

Reply:
> Yes but if the emulated code does actually print a newline then there will be a double newline.

So be it. If you are emulating a new line, than it must print the newline and that should not have anything to do with the additional newline that we print by defaul imo.

Reply:
Besides, the bash shell works like this and I'm just following bash in this regard.

Reply:
> Besides, the bash shell works like this

What do you mean?

```
ret2libc@fedora:~$ echo -n "Hello"
Helloret2libc@fedora:~$ echo -ne "Hello\n\n"
Hello

ret2libc@fedora:~$ 
```

Reply:
Exactly!

> So be it.

Ok.